src/pytorch_lightning/serve/servable_module_validator.py

```diff
@@ -0,0 +1,142 @@
+import time
+from multiprocessing import Process
+from typing import Any, Dict, Optional
+
+import requests
+import torch
+from typing_extensions import Literal
+
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import Callback
+from pytorch_lightning.serve.servable_module import ServableModule
+from pytorch_lightning.strategies import DDPFullyShardedNativeStrategy, DDPFullyShardedStrategy, DeepSpeedStrategy
+from pytorch_lightning.utilities.imports import _RequirementAvailable
+from pytorch_lightning.utilities.model_helpers import is_overridden
+from pytorch_lightning.utilities.rank_zero import rank_zero_only
+
+_NOT_SUPPORTED_STRATEGIES = (
+    DeepSpeedStrategy,
+    DDPFullyShardedNativeStrategy,
+    DDPFullyShardedStrategy,
+)
+
+
+class ServableModuleValidator(Callback):
+    """The ServableModuleValidator callback enables to validate a model is servable following the ServableModule
```

Comment:
Can you add the warning about this being experimental? Also, should we include it in the docs reference already?

Reply:
Yes, I am planning to add it to the docs with some examples.